### DISCUSSION TITLE
What are the best options to record a track during hiking?

### DISCUSSION THREAD

**Discussion by printmaps** [devemux86/cruiser#650] [Q&A] [answered]
I have recorded a track with Cruiser 5.0.0 (blue line), and parallel (on the same device) with Locus Map (red line). The cruiser track contains a lot of unexpected points.

<img width="2017" alt="Bildschirmfoto 2024-10-25 um 17 43 13" src="https://github.com/user-attachments/assets/ab99e531-b77b-4513-aaaa-3b58467de8ea">

<img width="2017" alt="Bildschirmfoto 2024-10-25 um 17 45 45" src="https://github.com/user-attachments/assets/970b9f8f-3c18-4bba-9f40-f0fb59c0926e">

**Comment by devemux86** [comment]
@printmaps Thanks for the detailed report.

Was Cruiser with open screen all the time?
Maybe you need to exclude Cruiser from battery optimization, please see here: https://dontkillmyapp.com

Also go in "Settings | GPS recording | Filters":
- You can decrease the "Minimum GPS accuracy", so only very accurate GPS locations are recorded.

- Are the spikes when you made stops? You could also set a "Minimum speed" for recording.

Does Locus have similar settings for GPS recording?

---

What have you selected in "Settings | Location | Location service"? Android or Google?

You can switch to a different location service and see if it works better on your device.

**Reply by devemux86** [reply]
@printmaps If you do more tests, please share the results here!

**Comment by printmaps** [comment]
> Was Cruiser with open screen all the time?

No

> Maybe you need to exclude Cruiser from battery optimization, please see here: https://dontkillmyapp.com/

Yes

Typical smartphone usage on a hike:
- Start hiking: start track recording, switch device off
- At (trail) junction: switch device on, orientate on map, switch device off
- At (trail) junction: switch device on, orientate on map, switch device off
- ...
- End hiking:  stop and save track recording, switch device off

> Does Locus have similar settings for GPS recording?

Yes, Locus Map has an option to avoid "automatic termination" (sorry, screenshot is in German):

![locus](https://github.com/user-attachments/assets/d99da010-8377-426a-8069-d0b94c1cfd01)

Run Locus Map Classic as a service
Secure the app against automatic termination by the system

**Reply by devemux86** [reply]
> Does Locus have similar settings for GPS recording?

I asked if Locus has settings for GPS recording, like the filters in Cruiser.

---

Cruiser automatically shows a location notification in navigation and GPS recording,
as Android mandates, so the system must respect that and allow to work in background or screen closed.

But OEMs change Android and apply aggressive battery optimizations when apps are in background or screen closed.
That's why we must remove these apps from the battery optimization, as described in https://dontkillmyapp.com

This is a standard process for all apps, it is recommended in OsmAnd forum, etc.

**Comment by airald** [comment]
To determine if the problem is related to power saving settings maybe could be interesting compare both tracks point by point and see what happens on locus when cruiser record a wrong point. If there are no points on locus where cruiser records a wrong point, the problem is the filter parameters.

I record tracks usually when hiking with Cruiser without problems like that. My filter settings are:
Time interval 5s
Min distance 5m
GPS min accuracy 50m
Min speed 1 km/h 

If you have raw tracks you can filter them using applications like GPS track editor. It has different algorithms in order to delete wrong points.

**Comment by printmaps** [comment]
These are the track recording settings in Cruiser and Locus Map:

Cruiser GPS recording settings:
5s - 10m - 50m
Minimum speed: off (maybe modified during track recording)
Map follow mode: off

Locus Map GPS recording settings:
10s - 1m - 100m
Map follow mode: on

The recording device (smartphone) has Android 14 with OneUi 6.1. According to 'dontkillmyapp':

UPDATE 07/2024: Samsung officially promised to drop the non-standard optimizations:
"To strengthen the Android platform, our collaboration with Google has resulted in a unified policy that we expect will create a more consistent and reliable user experience for Galaxy users. Since One UI 6.0, foreground services of apps targeting Android 14 will be guaranteed to work as intended so long as they are developed according to Android's new foreground service API policy.”

Recorded tracks (for further analysis): [Archiv.zip](https://github.com/user-attachments/files/17529064/Archiv.zip)

**Reply by devemux86** [reply]
This is a nice promise, but we can not know the result and how they still change Android.

It is always strongly recommended to exclude apps from battery optimization,
if we want the apps to work correctly in the background or with closed screen.

> they are developed according to Android's new foreground service API policy

Cruiser has been developed with Android's official guidelines from the beginning.

**Comment by printmaps** [comment]
A track compare with RouteConverter shows this:

<img width="1407" alt="Bildschirmfoto 2024-10-26 um 07 02 52" src="https://github.com/user-attachments/assets/d3669661-8be3-40af-a108-16e45b9dbd9c">

<img width="1407" alt="Bildschirmfoto 2024-10-26 um 07 03 27" src="https://github.com/user-attachments/assets/4c30e73c-929c-4c6f-a4bc-2ab03fbc4f02">

Remarks concerning the track recorded with Cruiser:
- Time stamps missing 
- Track to long (23.7 km instead of 5.8 km, correct is 5.8 km)
- Altitude information different (100m instead of 60m, correct is 60m)

**Reply by devemux86** [reply]
@printmaps 

> Time stamps missing

In the export screen you can select the GPX elements, time is included if you select it.
(it is selected by default)

> Altitude information different (100m instead of 60m, correct is 60m)

Select "Mean sea level" in "Settings | Location | Advanced | Altitude".

<img src="https://github.com/user-attachments/assets/658106bc-f342-4121-9610-33a01dfb79c0" width="50%"/><img src="https://github.com/user-attachments/assets/6b18dd26-2eda-470b-a748-fbc184ca6aff" width="50%"/>

**Comment by devemux86** [comment]
@printmaps 

What have you selected in "Settings | Location | Location service"? Android or Google?

You can switch to a different location service and see if it works better on your device.

**Comment by printmaps** [comment]
I recorded a new track which looks much better. It only contains one 'spike'. This one:

<img width="1351" alt="Bildschirmfoto 2024-10-27 um 17 25 44" src="https://github.com/user-attachments/assets/03a8bbd8-5093-43b5-b8db-dc1ed1c8d5e5">

Observations:

- GPS positions are recorded every 10 seconds
- exception: at 14:35:12 two positions are recorded

Here are more details:
```
      <trkpt lat="51.95469531394482" lon="7.523261962773383">
        <ele>120.0</ele>
        <time>2024-10-27T13:34:42Z</time>
        <hdop>2.5</hdop>
      </trkpt>
      <trkpt lat="51.95478032232925" lon="7.523426603337024">
        <ele>119.0</ele>
        <time>2024-10-27T13:34:52Z</time>
        <hdop>2.5</hdop>
      </trkpt>
      <trkpt lat="51.95484583170598" lon="7.523569738903481">
        <ele>120.0</ele>
        <time>2024-10-27T13:35:02Z</time>
        <hdop>2.5</hdop>
      </trkpt>
      <trkpt lat="51.9540311" lon="7.5236219">
        <ele>119.19999694824219</ele>
        <time>2024-10-27T13:35:12Z</time>
        <hdop>42.47100067138672</hdop>
      </trkpt>
      <trkpt lat="51.9549485296011" lon="7.523735845461488">
        <ele>119.0</ele>
        <time>2024-10-27T13:35:12Z</time>
        <hdop>2.5</hdop>
      </trkpt>
      <trkpt lat="51.95505424825945" lon="7.523883927284608">
        <ele>119.0</ele>
        <time>2024-10-27T13:35:22Z</time>
        <hdop>2.5</hdop>
      </trkpt>
      <trkpt lat="51.95518981136425" lon="7.523960676588472">
        <ele>121.0</ele>
        <time>2024-10-27T13:35:32Z</time>
        <hdop>2.5</hdop>
      </trkpt>
      <trkpt lat="51.9553131104168" lon="7.524040163015019">
        <ele>121.0</ele>
        <time>2024-10-27T13:35:42Z</time>
        <hdop>2.5</hdop>
      </trkpt>
      <trkpt lat="51.955438642215555" lon="7.5241194592638685">
        <ele>120.0</ele>
        <time>2024-10-27T13:35:52Z</time>
        <hdop>2.5</hdop>
      </trkpt>
      <trkpt lat="51.955450374477316" lon="7.524118303084078">
        <ele>121.0</ele>
        <time>2024-10-27T13:36:02Z</time>
        <hdop>2.5</hdop>
      </trkpt>
```

And that is the spike:

```
      <trkpt lat="51.9540311" lon="7.5236219">
        <ele>119.19999694824219</ele>
        <time>2024-10-27T13:35:12Z</time>
        <hdop>42.47100067138672</hdop>
      </trkpt>
      <trkpt lat="51.9549485296011" lon="7.523735845461488">
        <ele>119.0</ele>
        <time>2024-10-27T13:35:12Z</time>
        <hdop>2.5</hdop>
      </trkpt>
```
Assumption: The track recorder:

- detects the GPS inaccurcy (hdop = 42.47)
- requests the current position once again
- saves both points

**Reply by devemux86** [reply]
@printmaps Thanks for the detailed feedback.

It does not work like that, it only listens to GPS reports.
Then it filters them based on the selected filters.

You have not answered my question above:

>What have you selected in "Settings | Location | Location service"? Android or Google?
>You can switch to a different location service and see if it works better on your device.

**Reply by devemux86** [reply]
Where is the double location in the mentioned gpx text?

**Comment by printmaps** [comment]
> What have you selected in "Settings | Location | Location service"? Android or Google?

Android

> Where is the double location in the mentioned gpx text?

The timestamp occurs two times not the location:

Point 'A' (Position 321 in RouteConverter) at '2024-10-27T13:35:12Z' (hdop = 42.47, the 'spike'):

```
     <trkpt lat="51.9540311" lon="7.5236219">
        <ele>119.19999694824219</ele>
        <time>2024-10-27T13:35:12Z</time>
        <hdop>42.47100067138672</hdop>
      </trkpt>
```

Point 'B' (Position 322 in RouteConverter) also at '2024-10-27T13:35:12Z' (hdop = 2.5, the correct position):
```
      <trkpt lat="51.9549485296011" lon="7.523735845461488">
        <ele>119.0</ele>
        <time>2024-10-27T13:35:12Z</time>
        <hdop>2.5</hdop>
      </trkpt>
```

**Reply by devemux86** [reply]
@printmaps What accuracy do you have selected in GPS recording filter settings? The default 50m?

If your device has so good accuracy 2.5m, you can select a much lower accuracy 10-20m to filter out the errors.

**Comment by devemux86** [comment]
We could change the default filter accuracy setting from 50m to a lower value?

**Comment by printmaps** [comment]
> What accuracy do you have selected in GPS recording filter settings? The default 50m?

10s - 1m - 50m (min speed off)

```
      <trkpt lat="51.9540311" lon="7.5236219">
        <ele>119.19999694824219</ele>
        <time>2024-10-27T13:35:12Z</time>
        <hdop>42.47100067138672</hdop>
      </trkpt>
      
      <trkpt lat="51.9549485296011" lon="7.523735845461488">
        <ele>119.0</ele>
        <time>2024-10-27T13:35:12Z</time>
        <hdop>2.5</hdop>
      </trkpt>
```

The distance between the above points (the 'spike') is 102.4 meters.

Remarks:

- The time difference between two recorded points is typical 10 seconds.
- But there are some exceptions with a time difference of 0 seconds.
- The above points, the 'spike', is such an exception.

Questions:

- Why are (unwanted) points with the same timestamps recorded?
- How does the current filter work?
- Why has the filter not detected the 'spike'?

**Reply by devemux86** [reply]
Detect spike, how? Do you mean a new filter for "Maximum distance"?
If it is more than max distance, then don't record the GPS report.

**Comment by printmaps** [comment]
> There will be an improvement here in the next update for the same timestamps.
However the first GPS report will be recorded because it passes the selected filters.

Defect 1 - multiple GPS values for same timestamp:
Recommendation: If you get more than one GPS value for the same timestamp, record the most accurate value. That is the one with the smallest hdop value. In the above case the second GPS value with 'hdop=2.5'.

Defect 2 - general 'spike' detection and filtering:
To record 'smooth' and realistic tracks is a sophisticated task. Garmin is best at this, but treats it's algorithm as secret. We should open a new discussion for this subject. Best after fix for 'defect 1' is available.

**Reply by 0709wiwiwi** [reply]
The “wrong” Cruiser record after a treatment using the gps track editor.
http://www.gpstrackeditor.com/
How the gps track editor handles this no idea, not studied that yet.
Just for information, possibly it helps a little to optimize ?
[Aasee-Runde-Cruiser_by gps track editor.gpx.txt](https://github.com/user-attachments/files/17543606/Aasee-Runde-Cruiser_by.gps.track.editor.gpx.txt)

**Comment by printmaps** [comment]
@0709wiwiwi : GPS-Track-Editor is unfortunately Windows-only software, so I can't test it by myself. But the results are astonishing.

The (unmodified) original track:
<img width="1482" alt="Bildschirmfoto 2024-10-28 um 19 14 03" src="https://github.com/user-attachments/assets/47c37d3b-8d4e-44f8-9aa0-75ed3ac320eb">

The modified track by GPS-Track-Editor:
<img width="1482" alt="Bildschirmfoto 2024-10-28 um 19 13 45" src="https://github.com/user-attachments/assets/a8d1bf30-364b-4cee-a0a1-f608cd0fdb06">

Remarks:

- The modified track looks very 'smooth' and very realistic.
- The modified track contains no 'spikes' anymore.
- GPS-Track-Editor reduced the GPS points from 632 to 517.
- GPS-Track-Editor inserted fake timestamps.

**Comment by devemux86** [comment]
@printmaps 

Do you have the recorded gpx track by Cruiser with all data, timestamps and accuracy?
You can export it with the GPS recording menu.

**Comment by printmaps** [comment]
> Do you have the recorded gpx track by Cruiser with all data, timestamps and accuracy?

Currently not. Cruiser 5.0.0 exports from internal track management (database?) only the location and elevation data, but not the timestamps and not the accuracy data (hdop).

**Reply by devemux86** [reply]
This is possible today when you export from the GPS recording menu.

@printmaps You mentioned [above](https://github.com/devemux86/cruiser/discussions/650#discussioncomment-11067206) double timestamp, isn't this GPX available?

I can improve the filtering, but I need a GPX with all data to analyze it.
Because with my devices I cannot reproduce this.

**Comment by printmaps** [comment]
> This is possible today when you export from the GPS recording menu.

I only have the track 'Aasee-Runde-Cruiser' in the 'internal track database'. From there an export with timestamps and accuracy infos is not possible with Cruiser 5.0.0. But I have another track with multiple coords for the same timestamp and one big 'spike'. This track was exported immediate after track recording (and bevor deleting). I could provide this one.

**Reply by devemux86** [reply]
> But I have another track with multiple coords for the same timestamp and one big 'spike'. This track was exported immediate after track recording (and bevor deleting). I could provide this one.

@printmaps Thanks, we need anything that would help. :+1:

**Comment by printmaps** [comment]
Here is the track. The 'spike' is marked in the screenshot.

<img width="1399" alt="Bildschirmfoto 2024-10-30 um 09 54 14" src="https://github.com/user-attachments/assets/88e68b9e-84a2-45fd-9dae-85949eb33f2c">

[One-Spike-20241027_090514-2.gpx.zip](https://github.com/user-attachments/files/17568813/One-Spike-20241027_090514-2.gpx.zip)

**Reply by devemux86** [reply]
@printmaps Thanks! :+1:

**Reply by 0709wiwiwi** [reply]
The gps track editor sees even more errors.
Track points with an identical timestamp but slightly different cooordinates.  These locations are shown and automatically removed (if the user agrees).  The export file is attached.
[One-Spike-20241027_090514-2.gpx.txt](https://github.com/user-attachments/files/17575251/One-Spike-20241027_090514-2.gpx.txt)

**Reply by devemux86** [reply]
@0709wiwiwi Can you provide details or screenshot with these points?

**Reply by 0709wiwiwi** [reply]
![Screenshot](https://github.com/user-attachments/assets/0de3a315-1615-44a9-aa1a-5d566fc56a37)

**Comment by devemux86** [comment]
@printmaps 

Did you have the "Kalman filter" enabled in the location / advanced settings during your above recordings?

**Comment by printmaps** [comment]
I have recorded two new tracks with Cruiser 5.0.1. Both tracks are looking realistic. Both tracks have no multiple points for the same timestamp anymore. One track was recorded with kalman filter on, the other one without kalman filter.

**Reply by devemux86** [reply]
@printmaps 

Did you have the "Kalman filter" enabled during your reported recordings with the double points?

**Comment by printmaps** [comment]
> Did you have the "Kalman filter" enabled during your reported recordings with the double points?

Track around lake (many spikes): no kalman filter
Track with one spike: unclear

**Comment by printmaps** [comment]
Problem with multiple GPS points for same timestamp not solved.

I have a new track with many duplicate GPS points for same timestamp. BTW: Kalman filter for GPS location was on. Recording was done with Cruiser 5.0.1.

<img width="219" alt="Bildschirmfoto 2024-11-01 um 11 17 45" src="https://github.com/user-attachments/assets/468a63bc-c08e-430a-b257-807c0a878e9a">

**Reply by devemux86** [reply]
No improvements here were announced in the releases.
I still try to figure out what's going on, because it can't be reproduced here.

Kalman filter could cause this in GPS recording, but it's not really necessary here.
As only the native GPS reports should be recorded.
So it can be excluded from GPS recording in the future.

**Comment by printmaps** [comment]
> No improvements here were announced in the releases.

OK. I was confused by this comment:

> There will be an improvement here in the next update for the same timestamps.
> Available in 5.0.1-b14.

How was the above mentioned track recorded?
I put the GPS device for 2.5 hours (2:39) on the sofa (no movement). After this time the following distance was calculated:

- Cruiser = 1801 m
- Locus = 393 m

Could the problem 'multiple GPS points for same timestamp' be related to no movement?

**Reply by devemux86** [reply]
> Could the problem 'multiple GPS points for same timestamp' be related to no movement?

To compare the distances we must see the 2 gpx files to check the content.
If there are spikes, then the distance could be larger.

**Reply by devemux86** [reply]
Also what recording filters each app uses.
What location service, Android or Google.
Kalman filter or not.

We need all location + recording settings to check the 2 distances.

**Comment by printmaps** [comment]
> Also what recording filters each app uses.
What location service, Android or Google.
Kalman filter or not.

Cruiser: 10s - 1m - 50m (kalman filter on)
Locus: 10s - 1m - 100m (filter ?)

@0709wiwiwi : Could you try to reproduce on your device?

**Reply by devemux86** [reply]
@printmaps Thanks, can you try the next tests without the Kalman filter?

**Reply by 0709wiwiwi** [reply]
@printmaps 
I rarely use and keep trackrecords.  Cruiser settings ? 
Time interval 5 s, min distance 10m, GPS precision 50m, min speed off, no Kalman, service Android.
Anway I still found a few Cruiser trackrecords ;-)

Along the way I had accidently briefly turned off the phone so 2 trkseg.
[Steinbergbahn_mittelstation_20240906_140956.gpx.txt](https://github.com/user-attachments/files/17600932/Steinbergbahn_mittelstation_20240906_140956.gpx.txt)

With a clear sky view, trackrecords as well with Locus or Cruiser are just fine. 
[Asitz-Biberg_20240902_182222.gpx.txt](https://github.com/user-attachments/files/17605983/Asitz-Biberg_20240902_182222.gpx.txt)

**Comment by printmaps** [comment]
@0709wiwiwi : Thanks for testing and providing the track. If possible try this: Just start track recording and put the GPS device for 3 hours somewhere in your flat. Don't move it. This reproduces the problem 'multiple GPS points for same timestamp' for me on two different smartphone.

PS: For private reasons don't share the track here, just inform us about the result.

**Reply by devemux86** [reply]
@printmaps 

If the distance filter is enabled, it should not record locations, as the device is not moved.
Unless the GPS jumps around the real location.

**Reply by 0709wiwiwi** [reply]
Somewhere IN is usually the trigger for creating such "blobs".
Halted IN a building,  phone IN a pocket, IN a forest etc.
Once as a test I ran a record every second for 2 hours with the phone on a garden table . 
Result was a single deviation but no real “blob cloud”.

Stop the record by reading the accelaration sensor have been posted in forums. 
But the experiments so far have not been succesfull if I am correctly informed.

Removing or preventing track record blobs seems to be a popular subject ;-)
https://help.locusmap.eu/topic/36286-edit-track-cut-inner-part-and-leave-a-waypoint-with-time-delta

**Comment by printmaps** [comment]
> Unless the GPS jumps around the real location.

GPS typically jumps around, you get a "point cloud" (scatter plot). Just try it out.

**Reply by devemux86** [reply]
@printmaps Does it jump around more than the selected distance in the filters?

**Reply by WolfgangZ42** [reply]
I don't quite understand the whole discussion. So far I haven't had any problems with inaccuracies in GPS recording, especially when hiking. This afternoon I recorded a walk - w/o any problem: 
![2024-11-01 23_24_55-CF_20241101_142823 gpx - RouteConverter OpenSource Edition 3 1 vom 2024 Jul 27](https://github.com/user-attachments/assets/41d35981-0af7-434e-aa1e-6de4996dd1f8)

And the jump test showed a maximum deviation of 17m:
![2024-11-01 23_08_51-20241101_225727 gpx - RouteConverter OpenSource Edition 3 1 vom 2024 Jul 27](https://github.com/user-attachments/assets/42741ec3-4e4e-47da-bf1f-446dc6b56fb0)

Okay - my patience ended after 18 minutes ...

In both cases location service = **Android**, Kalmanfilter **On** and with following filters:
![image](https://github.com/user-attachments/assets/ec1de8d5-ce1d-4096-b44b-b1953387b5ed)

**Comment by printmaps** [comment]
@WolfgangZ42

> I don't quite understand the whole discussion.

What is the defect?
Example: Cruiser should record GPS points every ten seconds. Under unclear circumstances Cruiser sometimes records two GPS points for the same timestamp (this should never happen). The coords of these two GPS points are sometimes (not often) more or less different. This leads to an unwanted and incorrect extra distance.

How to force / reproduce the defect?
The behavior can be reproduced by recording a track indoors (without any movement of the GPS device).

Do you also have the defect?
Yes. Your second screenshot shows the defect. You recorded GPS points for the same timestamp (see Position 3+4 and Position 5+6).

**Reply by devemux86** [reply]
The same timestamps will be fixed in the next update.

The spikes depend on the GPS accuracy of each device.
That's why filters exist, we can decrease the accuracy filter to e.g. 5m and record only the very accurate GPS locations.

**Comment by devemux86** [comment]
**[Cruiser 5.0.2-b1](https://github.com/devemux86/cruiser/discussions/641#discussioncomment-11134244)** contains GPS recording improvements and uses only the native GPS providers.

**Comment by 0709wiwiwi** [comment]
Phone placed on the garden table. 
Both apps no location filter, no Google. 
Compass test, heading indicator, trackrecord stability.
Locus 4.26.3 vs Cruiser 5.02.

Locus map is still possible in small steps.
Locus location and track records stable.
Locus heading indicator is optimal for hiking.
Locus vector map rotate with the compass.

**Reply by devemux86** [reply]
@0709wiwiwi Thanks for the video.

You didn't mention the settings of the test, location service, recording filters, etc.

> location and track records are stable see video.

We do not snap / modify the location, it is the original location as reported by device sensor.
GPS recording filters are there to select how much filtering you can apply on the recording.

> heading compass indicator is optimal for hiking.
> vector map can also rotate using the compass.

Probably you don't follow closely Cruiser features? Everything is already available:
- https://github.com/devemux86/cruiser/discussions/628
- https://github.com/devemux86/cruiser/discussions/625#discussioncomment-10544949

**Reply by 0709wiwiwi** [reply]
> Probably you don't follow closely Cruiser features?

Correct. So I just now switched to v 5.0 this to test that trackrecord stability report.
So map rotation by the compass found and is set  OK ;-)
- Small zoom steps no ?  Heading Line no ?  
Position stability ?  Cruiser location shows jumps while Locus does not.
Both apps same test are operating in split screen.
Probably @printmaps can run that same test by splitted screen operation ?

**Reply by devemux86** [reply]
The location is displayed at the exact coordinates where the GPS sensor reports it.
Why should we place it elsewhere or filter it, just for aesthetic reasons?

Whether Locus filters the GPS location all the time, I don't know, you probably know better?

Also try a different location service in Cruiser, maybe it works better on your device.

**Reply by devemux86** [reply]
Also you didn't answer the question about the settings.
What location service, Kalman filter is on or off, etc.

**Reply by 0709wiwiwi** [reply]
Cruiser settings: No filter no Google.
Locus settings. No filter no Google.
What strikes me is the live location jumps seen in Cruiser while Locus does not show.

**Reply by devemux86** [reply]
Settings do not say what each app does internally.

Try a different location service in Cruiser location settings. Maybe Google works better on your device.
App shows the real location changes by the sensor, this is important in navigation and follow modes.

**Comment by printmaps** [comment]
What exactly is the difference between "location service Google" and "location service Android"?

**Reply by devemux86** [reply]
- "Android" uses the Android provider (open source)
- "Google" uses the [Google fused provider](https://developers.google.com/location-context/fused-location-provider)

Can try both and see what works better on each device.

**Comment by printmaps** [comment]
We must distinguish between two situations:

**Defect - duplicate GPS points for the same timestamp:**
This problem should never happen but occurs occasionally and is classified as a defect in Cruiser. There is a correction for this in version 5.0.2.

**Optimization - less error dispersion:**
As a rule, a GPS signal is imprecise. This effect is particularly noticeable if the GPS device is not moved for a longer period of time.
The GPS signal bounces back and forth, creating a point cloud. The (good) optimization of a recorded track is a challenging task. The definition of simple filter conditions is not sufficient. The (post) optimization must be done on the unmodified, unfiltered raw GPS data given from the GPS device.

**Comment by 0709wiwiwi** [comment]
I had a test by both Cruiser and Locus . Only visually observing the location stability.

Location  Filter OFF/
(The Kalman filter has no real influence on the location jumps in Cruiser.)

- Cruiser Android: Gps jumps YES.
- Locus Android : Gps jumps NO

- Cruiser Google: Gps jumps NO
- Locus Google : Gps jumps NO

Split screen compare Locus vs Cruiser.
- Locus Android 
- Cruiser Google. 
The location between both app displays is a little different.

Split screen compare Locus vs Cruiser.
- Locus Google 
- Cruiser Google.
The location between both app displays is identical.

**Reply by devemux86** [reply]
@0709wiwiwi Thanks for the detailed test.

We could set Google as default location service (in new installs).
Also OsmAnd like Locus has Google as default.

**Comment by 0709wiwiwi** [comment]
@printmaps 
Information I read you are also using Locus map.
If interested in better gps quality receivers read and eventually contact Locus user Olaf (region Aachen-Eifel) by the Locus forum PM messenger for extra detailled information about his interesting home made project.  Best Regards.
https://forum.locusmap.eu/index.php?topic=8944.msg75828#msg75828

**Comment by devemux86** [comment]
- https://github.com/devemux86/cruiser/discussions/641 has GPS recording improvements

---

Remember to select "**Google Play Services**" as location service (now default)
and adjust the GPS recording filters, especially the "Maximum GPS accuracy".

**Reply by bios4** [reply]
> adjust the GPS recording filters, especially the "Maximum GPS accuracy"

What are the recommended settings for these @devemux86?

**Reply by devemux86** [reply]
@bios4 OsmAnd has 50m, Locus has 100m. It depends on the GPS sensor.

You can record a track, export it with accuracy as GPX and check the `hdop`.
A general rule of thumb is that a lower accuracy filter will have fewer spikes.

**Comment by Klaus-Tockloth** [comment]
During initial tests with Cruiser v5.0.4, only one case of ‘duplicate GPS data’ occurred for one timestamp. The problem does not seem to have been solved yet completely. Further tests will follow.

**Reply by WolfgangZ42** [reply]
This discussion seems to be getting quite “academic” - I'm out ...
BTW: In the last few days I had recorded a bike tour with ~24 km and a hike with ~8 km out of pure curiosity and did not see any “duplicates” in the RouteConverter.

**Reply by devemux86** [reply]
@Klaus-Tockloth Please attach the exported GPX track to check it.

What location service was selected, Google or Android?

**Comment by printmaps** [comment]
The attached GPX track (recorded with Cruiser v504) shows the defect multiple times.

<img width="1265" alt="Bildschirmfoto 2024-11-10 um 18 11 16" src="https://github.com/user-attachments/assets/cba93e24-6e47-454a-bec5-8c61078c7feb">

[Cruiser-v504-20241110_155828.gpx.zip](https://github.com/user-attachments/files/17692339/Cruiser-v504-20241110_155828.gpx.zip)

**Reply by devemux86** [reply]
@printmaps Thanks for the gpx file.

What location service was selected, Google or Android?

**Comment by printmaps** [comment]
> What location service was selected, Google or Android?

Google Play Services

**Comment by devemux86** [comment]
I found an improvement in GPS recording, we could test it in the next Beta.

**Comment by printmaps** [comment]
> I found an improvement in GPS recording, we could test it in the next Beta.

That sounds promising.

**Reply by devemux86** [reply]
The GPS recording improvement is available for testing in:
- https://github.com/devemux86/cruiser/discussions/671

**Comment by devemux86** [comment, marked as answer]
- GPS recording improvements are available in https://github.com/devemux86/cruiser/discussions/684
